tests/test_lab_evaluate.py

```diff
@@ -0,0 +1,329 @@
+# Standard
+from unittest import mock
+from unittest.mock import patch
+
+# Third Party
+from click.testing import CliRunner
+
+# First Party
+from instructlab import lab
+
+
+def gen_qa_pairs(odd):
+    i = 1
+    qa_pairs = []
+    score = 0
+    while i < 5:
+        if i % 2:
+            if odd:
+                score = 0.2
+            else:
+                score = 0.1
+        elif not i % 2:
+            if odd:
+                score = 0.3
+            else:
+                score = 0.4
+        qa_pairs.append(
+            {
+                "question_id": i,
+                "score": score,
+                "qna_file": f"category{i}/qna.yaml",
+            }
+        )
+        i = i + 1
+    qa_pairs.append(
+        {
+            "question_id": i,
+            "score": 0.5,
+            "qna_file": f"category{i}/qna.yaml",
+        }
+    )
+    if odd:
+        qa_pairs.append(
+            {
+                "question_id": i + 1,
+                "score": 0.6,
+                "qna_file": f"category{i+1}/qna.yaml",
+            }
+        )
+    return qa_pairs
+
+
+def gen_individual_scores(odd):
+    i = 1
+    individual_scores = {}
+    score = 0
+    while i < 5:
+        if i % 2:
+            if odd:
+                score = 0.2
+            else:
+                score = 0.1
+        elif not i % 2:
+            if odd:
+                score = 0.3
+            else:
+                score = 0.4
+        individual_scores[f"task{i}"] = {"score": score}
+        i = i + 1
+    individual_scores[f"task{i}"] = {"score": 0.5}
+    return individual_scores
+
+
+class TestEvaluate:
```

Comment:
pytest does not need a class for tests. You could drop the class and move all methods to module-level functions.

Reply:
I was following the pattern from other tests.  Didn't know it wasn't a good one.  Fixed!

Reply:
In the beginning we used `unittest.TestCase`. When we moved to 100% pytest, I removed `unittest.TestCase` as parent class, but did not move all test methods to test functions. In pytest, test classes are not necessary but sometimes useful. In your case, you didn't need it and are now benefiting from 4 spaces less per line :)